### ISSUE TITLE
delete number cause cannot format correctly

### ISSUE THREAD

**Issue by vinurs** [caseyryan/flutter_multi_formatter#59]
![Kapture 2021-08-17 at 10 29 08](https://user-images.githubusercontent.com/37489300/129654490-1dcdf206-88d9-4693-8ea0-a68b9a63b6f4.gif)
when delete the number, then the formatter can not format the input correctly

**Comment by caseyryan** [comment]
what does +86 mean in the beginning? My formatter does not support phone numbers without country code. If you want to enter custom country code (like +86 in your case) it won't work. The formatter detects the code automatically and adds it on the fly

**Comment by vinurs** [comment]
i just use
```
inputFormatters: [
                MaskedInputFormatter("### #### ####"),
              ],
```

**Comment by caseyryan** [comment]
> i just use
> 
> ```
> inputFormatters: [
>                 MaskedInputFormatter("### #### ####"),
>               ],
> ```

Ah, ok. I thought you're talking about PhoneInputFormatter. 
I'll take a look at this issue as soon as I have time. Thanks for reporting

**Comment by caseyryan** [comment]
Fixed in version 2.3.5.

**Comment by vinurs** [comment]
> Fixed in version 2.3.5.

thanks very much

**Comment by vinurs** [comment]
![Kapture 2021-09-02 at 00 47 46](https://user-images.githubusercontent.com/37489300/131711619-f5b7b5ee-fd56-41fa-af11-25c852de8022.gif)

@caseyryan there is still a error, when i delete number, type =3 3 5 9= but the cursor =9= in front of =5=

**Comment by caseyryan** [comment]
> ![Kapture 2021-09-02 at 00 47 46](https://user-images.githubusercontent.com/37489300/131711619-f5b7b5ee-fd56-41fa-af11-25c852de8022.gif)
> 
> @caseyryan there is still a error, when i delete number, type `3 3 5 9` but the cursor `9` in front of `5`

Sorry for the delay. I was too busy at work. I took a look at your last report. Yes, that was kind of a tricky one. Fixed in version 2.3.7
